### ISSUE TITLE
Not getting proper response for error case

### ISSUE THREAD

**Issue by ashoks237** [stagingrdkm/LISA#46]
###  Install Existing Application:

```
TEST CASE ID     :  LISA_07
DESCRIPTION      :  Check if able to install the existing application
#==============================================================================##-------------- Test Case Pre-Requisite ---------------#Pre Requisite : Install_Test_Application
Pre Requisite No : 1TEST STEP NAME   :  Check_Application_List
TEST STEP ID     :  1
PLUGIN API NAME  :  LISA.1.getList
EXPECTED VALUES  :  essos_sercom
TEST STEP STATUS :  SUCCESS
----------------- Result -----------------
APPSTATUS        :  TRUE#--------- [Pre-requisite Status] : SUCCESS ----------#Test Case Pre-requisite Status: SUCCESS
JSON Cmd :  {"params": {"category": "category", "version": "1.0", "appName": "essos_sercom", "url": "http://192.168.160.29:8003/essos.tar.gz", "type": "dac", "id": "essos_sercom"}, "jsonrpc": "2.0", "id": 2, "method": "LISA.1.install"}Response :  {u'jsonrpc': u'2.0', u'id': 2, u'error': {u'message': u'', u'code': 1003}}TEST STEP NAME   :  Install_Application
TEST STEP ID     :  1
PLUGIN API NAME  :  LISA.1.install
INPUT PARAMETER  :  category - category, version - 1.0, appName - essos_sercom, url - http://192.168.160.29:8003/essos.tar.gz, type - dac, id - essos_sercom
EXPECTED VALUES  :  already installed,1003
TEST STEP STATUS :  FAILURE
----------------- Result -----------------
ERROR            :  {u'message': u'', u'code': 1003}##--------- [TEST EXECUTION STATUS] : FAILURE  
```

**Comment by sverkoye** [comment]
@ashoks237 , yes, indeed the error messages are empty. After some investigation IMHO it looks like there is no easy solution. LISA functions return an error code but I don’t see a way to return the related message. wpeframework does fill in a message for error codes that it knows itself (the common error codes): https://github.com/rdkcentral/Thunder/blob/89968f149299ff5647cc4a4ab9d4f948c858ebff/Source/WPEFramework/PluginServer.h#L1559
https://github.com/rdkcentral/Thunder/blob/89968f149299ff5647cc4a4ab9d4f948c858ebff/Source/core/Portability.h#L835 

setAuxMetadata API is returning as wrong params
root@mediaclient-mediabox:~# curl http://localhost:9998/jsonrpc --header "Content-Type: application/json" --request POST --data '{"jsonrpc":"2.0","id":"1","method":"LISA.1.setAuxMetadata","params":{"id":"ESSOS_VC4","version":"1.0","type":"dac","auxMetadata":[{"key":"essos","value":"sample meta data"}]}}'
{"jsonrpc":"2.0","id":1,"error":{"code":1001,"message":""}}root@mediaclient-mediabox:~#
root@mediaclient-mediabox:~# root@mediaclient-mediabox:~# curl http://localhost:9998/jsonrpc --header "Content-Type: application/json" --request POST --data '{"jsonrpc":"2.0","id":"1","method":"LISA.1.setAuxMetadata","params":{"id":"ESSOS_VC4","version":"1.0","type":"dac","auxMetadata":{"key":"essos","value":"sample meta data"}}}'
{"jsonrpc":"2.0","id":1,"error":{"code":1001,"message":""}}root@mediaclient-mediabox:~#


**Check Installed Application Storage Details**
What exactly is the problem here?

**Pass Non-Existent Application**
I can update the implementation to report error code 1001 when specific app not found

**setAuxMetadata**
setAuxMetadata does not accept and array auxMetadata with key/values but just one key+value together with id, type and version in the same object. See https://github.com/rdkcentral/ThunderInterfaces/blob/76713d53d7386d164fca30ad1be7e94230d49a0e/jsonrpc/LISA.json#L571

But I see the documentation is wrong. I will update it to match json interface and implementation: https://wiki.rdkcentral.com/display/ASP/LISA#LISA-Methods

**Comment by sverkoye** [comment]
> Check Installed Application Storage Details What exactly is the problem here? => Can you please share details on what is the expected QuotaKB ?

Quota is not implemented yet. Part of phase 2 as mentioned on the wiki page.

> Pass Non-Existent Application I can update the implementation to report error code 1001 when specific app not found => Do you have any updated implementation which we can try and check? Please share.

I will make the fix. It is not yet available.

> But I see the documentation is wrong. I will update it to match json interface and implementation: https://wiki.rdkcentral.com/display/ASP/LISA#LISA-Methods => Did you got updated documentation?

Documentation was updated. Check the page history.
